### ISSUE TITLE
Deleted items should goto an archive before being permanently deleted

### ISSUE THREAD

**Issue by hamzahc1** [lightdash/lightdash#3872]
### Description

When you delete a chart or dashboard in Lightdash, it should actually goto an Archive (that means it no longer appears in searches etc, but it can be recovered unless you Permanently Delete it from the Archive. 

This is one of the features I like the most from Notion! 

### What problem does this solve?

It's often easy to create but hard to delete anything in a BI tool. If you could "softly" delete things by moving them to an Archive folder (and then periodically deleting the Archive) - it would be a much lower barrier to cleaning up your org's folder without worrying if folks' will lose their work. 

### Acceptance criteria

- [ ] content is archived for 30 days after being deleted, after which point it's deleted permanently
- [ ] if you "unarchive" content, then all scheduled deliveries related to that content are also un-archived. 
- [ ] if you "unarchive" a dashboard, then all charts created within dashboards are also un-archived. 

### Nice to have

- [ ] when a preview project is closed, changes that only exist in the preview should also be put in "archive" somehow in case the preview is closed by accident with pending changes to dashboards and charts

**Comment by jasonlanderson** [comment]
Yes this would be a great feature to remedy accidental deletions

**Comment by ZeRego** [comment]
There aren't unknowns about how to do it, but is one of those tasks where implementing is 10% and testing is 90%.
The solution for this is to have a "archived: boolean" column in the dashboards and charts tables.
The time-consuming part is that we need to update all the queries to those tables to filter out rows where archived is true. If this is not done properly, we will start seeing weird behaviour (aka bugs) across all features.

**Comment by jasonlanderson** [comment]
Thanks @ZeRego!
